kong/init.lua

```diff
@@ -975,11 +982,18 @@ function Kong.init_worker()
   end
 
   if kong.clustering then
-    kong.clustering:init_worker()
 
-    local cluster_tls = require("kong.clustering.tls")
+    -- full sync cp/dp
+    if not kong.sync then
+      kong.clustering:init_worker()
+    end
```

Comment:
If we enable kong.sync, then kong.clustering:init_worker() will not run, which makes kong.clustering.instance nil.

Then running this test case will faile: ` FAILED spec/02-integration/09-hybrid_mode/01-sync_spec.lua:493: CP/DP #version check #postgres relaxed compatibility check: CP and DP plugin version matches to major.minor, sync is allowed`, it trigger a runtime error:

```
2024/10/15 17:44:24 [error] 39861#0: *646 lua entry thread aborted: runtime error: ./kong/clustering/init.lua:72: attempt to index field 'instance' (a nil value)                                                                                               stack traceback:                                                                                                                coroutine 0:
        ./kong/clustering/init.lua: in function 'serve_cluster_listener'
        content_by_lua(nginx-kong.conf:104):2: in main chunk, client: 127.0.0.1, server: kong_cluster_listener, request: "GET /v1/outlet?node_id=55f8c537-3e11-4d74-9ae7-eec55851818e&node_hostname=xiaochens-MacBook-Pro.local&node_version=3.9.0 HTTP/1.1", host: "127.0.0.1:9005"
```

Reply:
This issue only exists in CP side, we cloud fix it later, may not be this PR.

Reply:
fixed in https://github.com/Kong/kong/pull/13746